snakemake_rules/Snakefile

```diff
@@ -0,0 +1,134 @@
+configfile: "/home/mei.wu/rna-seq/config/config.yaml"
+
+COMMON_DIR = config["output_dir"]
+
+
+rule end:
+    input:
+        COMMON_DIR + "/star_index/",
+        expand(COMMON_DIR + "/{sample}/star_aln/{sample}.Aligned.sortedByCoord.out.bam.bai", sample=config["read1"]),
+        expand(COMMON_DIR + "/{sample}/multiqc/report.html", sample=config["read1"]),
+
+rule fastqc:
+    input:
+        read1 = lambda wildcards: config["read1"][wildcards.sample],
+        read2 = lambda wildcards: config["read2"][wildcards.sample]
+    output:
+        qc_read1 = COMMON_DIR + "/{sample}/fastqc/{sample}_1_fastqc.zip",
+        qc_read2 = COMMON_DIR + "/{sample}/fastqc/{sample}_2_fastqc.zip"
+    params:
+        tmp_dir = config["tmp_dir"],
+        outpath = COMMON_DIR + "/{sample}/fastqc/"
+    threads: 15
+    shell:
+        "fastqc --threads {threads} --noextract --dir {params.tmp_dir} --outdir {params.outpath} {input.read1};"
+        "fastqc --threads {threads} --noextract --dir {params.tmp_dir} --outdir {params.outpath} {input.read2};"
+
+rule trim_galore_pe:
+    input:
+        read1 = lambda wildcards: config["read1"][wildcards.sample],
+        read2 = lambda wildcards: config["read2"][wildcards.sample]
+    output:
+        trimmed_1 = COMMON_DIR + "/{sample}/trim-galore/{sample}_1_val_1.fq.gz",
+        trimmed_2 = COMMON_DIR + "/{sample}/trim-galore/{sample}_2_val_2.fq.gz"
+    params:
+        outpath = COMMON_DIR + "/{sample}/trim-galore/"
+    shell:
+         "trim_galore --cores 15 --fastqc --gzip --output_dir {params.outpath} --paired {input.read1} {input.read2}"
+
+rule star_index:
+    input:
+         fa = config["genome"],
+         gtf = config["gtf"]
+    output:
+         directory(COMMON_DIR + "/star_index/")
+    threads: 36
+    shell:
+         'STAR --runThreadN {threads} '
+         '--runMode genomeGenerate '
+         '--genomeDir {output}'
+         '--genomeFastaFiles {input.fa} '
+         '--sjdbGTFfile {input.gtf} '
+         '--sjdbOverhang 99'
```

Comment:
@jemten `--sjdbOverhang ReadLength-1`

How would I go about finding out this? Do I look at the Sequence Length Distribution in FASTQC results?

Reply:
also, this is what the command would look like `STAR --runThreadN 1 --runMode genomeGenerate --genomeDir /home/mei.wu/rna-seq/output/star_index --genomeFastaFiles /home/proj/development/rare-disease/references_9.0/grch38_homo_sapiens_-assembly-.fasta --sjdbGTFfile /home/proj/development/rare-disease/references_9.0/grch38_gencode_annotation_-v34-.gtf --sjdbOverhang 99`

Reply:
Regarding the read length. All files produced here have a read length of 150. After trimming some of them might be shorter but that isn't important for this option. 

Looking at the command you probably want to use all available cores on the node so  `--runThreadN 36`. Otherwise the command looks good 🚀